### PR TITLE
Even more show methods for libgit2

### PR DESCRIPTION
**Pull request by kshyatt**
Not super sure how to test the `GitTreeEntry` and `GitIndex` ones.

### CODE REVIEW

**Comment by kshyatt** [comment]
Can someone review this?

**Comment by tkelman** [comment]
Looks reasonable but without
> how to test the `GitTreeEntry` and `GitIndex`

or hitting the other types of `GitReference` it's hard to be sure if the ccalls are working properly.

**Comment by kshyatt** [comment]
Yeah I agree.

File: base/libgit2/reference.jl [hunk 1]
```diff
@@ -99,6 +99,21 @@ function isremote(ref::GitReference)
...
+    println(io, "GitReference:")
+    if isremote(ref)
+        println(io, "Remote with name ", name(ref))
+    elseif isbranch(ref)
+        println(io, "Branch with name ", name(ref))
+        if ishead(ref)
+            println(io, "Branch is HEAD.")
+        else
+            println(io, "Branch is not HEAD.")
+        end
```
Comment (ararslan) [inline base/libgit2/reference.jl]: It's perhaps a little too "cute," but you could write this more succinctly as
```julia
println(io, "Branch is ", ishead(ref) ? "not " : "", "HEAD.")
```

**Comment by kshyatt** [comment]
I'm going to merge and leave a note that we need to test these types better anyway, and testing their `show` methods will be a part of that.

**Pull request merged**